src/converters/table-post-fixer.js

```diff
@@ -0,0 +1,258 @@
+ * The table cell with FOO contents will not expand to the body section so its `rowspan` attribute will be changed to `1` (and as
+ * the value `1` is a default value the `rowspan` attribute will be removed from the model).
+ *
+ * The table cell with BAZ contents will be in the first column of a table.
+ *
+ * ## Adding missing table cells
+ *
+ * The table post-fixer will insert empty table cells to equalize table rows lengths. The width of a table row is calculated by counting
+ * widths of table cells and widths of columns spanned by cells from rows above in a given row.
+ *
+ * In the above example the table row in body section of the table is narrower then the row from the header - it has two cells
+ * with default colspan (value of `1`). The header row has one cell with width = 1 and second withe width = 2.
+ * The table cell FOO does not expand beyond head section (and as such will be fixed in the first step of this post-fixer).
+ * The post-fixer will add a missing table cell to the row in body section of the table.
+ *
+ * The table from the above example will be fixed and rendered to the view as below:
+ *
+ *		<table>
+ *			<thead>
+ *				<tr>
+ *					<td rowspan="2">FOO</td>
+ *					<td colspan="2">BAR</td>
+ *				</tr>
+ *			</thead>
+ *			<tbody>
+ *				<tr>
+ *					<td>BAZ<td>
+ *					<td>XYZ<td>
+ *				</tr>
+ *			</tbody>
+ *		</table>
+ *
+ * **Note** The table post-fixer only ensures proper structure without deeper analysis of the nature of a change. As such it might lead
+ * to a structure which was not intended by user changes. In particular it will also fix undo steps (in conjunction with collaboration)
+ * in which editor content might not return to the original state.
+ *
+ * @param {module:engine/model/model~Model} model
+ */
+export default function injectTablePostFixer( model ) {
+	model.document.registerPostFixer( writer => tablePostFixer( writer, model ) );
+}
+
+// The table post-fixer.
+//
+// @param {module:engine/model/writer~Writer} writer
+// @param {module:engine/model/model~Model} model
+function tablePostFixer( writer, model ) {
+	const changes = model.document.differ.getChanges();
+
+	let wasFixed = false;
+
+	for ( const entry of changes ) {
+		let table;
+
+		// Fix table on table insert.
+		if ( entry.name == 'table' && entry.type == 'insert' ) {
+			table = entry.position.nodeAfter;
+		}
+
+		// Fix table on adding/removing table cells and rows.
+		if ( entry.name == 'tableRow' || entry.name == 'tableCell' ) {
+			table = getParentTable( entry.position );
+		}
+
+		// Fix table on any table's attribute change - including attributes of table cells.
+		if ( isTableAttributeEntry( entry ) ) {
+			table = getParentTable( entry.range.start );
+		}
+
+		if ( table ) {
+			// Step 1: correct rowspans of table cells if necessary.
+			wasFixed = fixTableCellsRowspan( table, writer ) || wasFixed; // Should be true if any of present tables was fixed.
```

Comment:
I guess ` || wasFixed` is not necessary in this line. It's always `false`, as defined in L117.

Reply:
You guessed wrong :) This case covers insertion of more then one table in one batch or inserting multiple table cells or rows in a batch.

In case that first table was fixed but the second one was not the `wasFixed` flag must be always `true` as there were fixes in whole batch.

ps.: I've updated the code a bit so a table will be analyzed only once in one batch.